apps/meteor/ee/app/livechat-enterprise/server/hooks/beforeNewInquiry.js

```diff
@@ -13,24 +13,33 @@ callbacks.add(
 			return extraData;
 		}
 
-		const sla = Promise.await(
-			OmnichannelServiceLevelAgreements.findOneByIdOrName(searchTerm, {
-				projection: { dueTimeInMinutes: 1 },
-			}),
-		);
-		if (!sla) {
-			throw new Meteor.Error('error-invalid-priority', 'Invalid sla', {
+		const slaP = OmnichannelServiceLevelAgreements.findOneByIdOrName(searchTerm, {
+			projection: { dueTimeInMinutes: 1 },
+		});
+		const priorityP = LivechatPriority.findOneByIdOrName(searchTerm, {
+			projection: { _id: 1, priorityWeight: 1 },
+		});
+		const [sla, priority] = Promise.await([slaP, priorityP]);
+		if ((!sla && !priority) || (sla && priority)) {
+			throw new Meteor.Error('error-invalid-priority', 'Invalid sla or priority', {
 				function: 'livechat.beforeInquiry',
 			});
 		}
```

Comment:
```suggestion
		if (!sla && !priority) {
			throw new Meteor.Error('error-invalid-priority', 'Invalid sla or priority', {
				function: 'livechat.beforeInquiry',
			});
		}
		if (sla && priority) {
			throw new Meteor.Error('error-invalid-state', 'You cannot have both sla and priority assigned to the same inquiry at a time', {
				function: 'livechat.beforeInquiry',
			});
		}
```

The second condition in the "IF" condition doesn't really fit in with the error message :eyes: Here's a possible suggestion for better error handling here